File: internal/client/roles.go
package client

import (
	"context"
	"fmt"
	"net/http"
)

// Role mirrors the Role schema (task 8.4/8.7) - one of the three fixed
// built-in roles (decision #30). Read-only: there is no create/update/
// delete, since roles aren't admin-editable.
type Role struct {
	ID          int64    `json:"id"`
	Name        string   `json:"name"`
	Description string   `json:"description"`
	Permissions []string `json:"permissions"`
}

type RoleList struct {
	Items []Role `json:"items"`
}

func (c *Client) ListRoles(ctx context.Context) (RoleList, error) {
	var list RoleList
	err := c.do(ctx, http.MethodGet, "/api/v1/roles", requestOptions{}, &list)
	return list, err
}

func (c *Client) GetRole(ctx context.Context, name string) (Role, error) {
	var role Role
	err := c.do(ctx, http.MethodGet, fmt.Sprintf("/api/v1/roles/%s", name), requestOptions{}, &role)
	return role, err
}

File: web/src/api/roles.ts
import { request } from './client'
import type { components } from './schema'

export type Role = components['schemas']['Role']
export type RoleList = components['schemas']['RoleList']

// List-only - roles are fixed built-ins (ARCHITECTURE.md §6 decision #30),
// not admin-editable, so there is no create/update/delete here.
export function listRoles(): Promise<RoleList> {
  return request<RoleList>('/api/v1/roles')
}
